# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/chrome-webidl/FrameLoader.webidl
# Commit: 909001ac06d8
# Full Hash: 909001ac06d8b3982116a0b0f753ae365b1cea22
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2020-07-23 03:20:25
# Regressor Bug: 1652270
# File Overlap Count: 1
# Description:
#   Bug 1652270. Convert nsFrameLoader::Print to return a Promise. r=farre,remote-protocol-reviewers,marionette-reviewers,jgraham,whimboo,mixedpuppy
#   
#   (Instead of requiring callers to pass an nsIWebProgressListener.)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83249
# ==============================================================================

diff -r 8cfbd5213d04 -r 909001ac06d8 dom/chrome-webidl/FrameLoader.webidl
--- a/dom/chrome-webidl/FrameLoader.webidl	Thu Jul 23 02:01:54 2020 +0300
+++ b/dom/chrome-webidl/FrameLoader.webidl	Wed Jul 22 23:17:45 2020 +0000
@@ -122,12 +122,11 @@
    * @param aOuterWindowID the ID of the outer window to print
    * @param aPrintSettings optional print settings to use; printSilent can be
    *                       set to prevent prompting.
-   * @param aProgressListener optional print progress listener.
+   * @return A Promise that resolves once printing is finished.
    */
   [Throws]
-  void print(unsigned long long aOuterWindowID,
-             nsIPrintSettings aPrintSettings,
-             optional nsIWebProgressListener? aProgressListener = null);
+  Promise<void> print(unsigned long long aOuterWindowID,
+                      nsIPrintSettings aPrintSettings);
 
   /**
    * The element which owns this frame loader.